### PR TITLE
updated development workflow and repatriation removal

### PR DESCRIPTION
**Pull request by LanceEa**
## Description

This PR includes commits for updating our docs to include the new development workflow and contributing guidelines as outlined in the related issue. 

- update DEVELOPING.md with new guidelines on the development workflow, branching and merging strategies and the Pull Request review process
- remove repatriation GIthub Action
- update pull request template

**Note**: this will need to be backported to the `release/v2.5` branch since this is the latest minor branch of the previous version.  I will also update the `release/v1.14` to remove the repatriation action which I believe is in circle-ci. 

## Related Issues
#4614

## Testing
N/A

## Checklist
 - [ ] I made sure to update `CHANGELOG.md`.
 - [x] This is unlikely to impact how Ambassador performs at scale.
 - [ ] My change is adequately tested.
 - [x] I updated `DEVELOPING.md` with any any special dev tricks I had to use to work on this code efficiently.
 - [x] The changes in this PR have been reviewed for security concerns and adherence to security best practices.

### CODE REVIEW

**Comment by LanceEa** [comment]
@emissary-ingress/emissary-maintainers - Are there any outstanding PR's in flight that want to take advantage of the machinery before this is merged?

@LukeShu - I think this is the only one I'm aware of at this time. https://github.com/emissary-ingress/emissary/pull/4055

**Review by ddymko — DISMISSED** [review]

File: DEVELOPING.md
Comment (haq204) [inline DEVELOPING.md:213]: Don't think make image and make push was mentioned beforehand. Maybe just reword to say these targets exists if you want to do individual things

File: DEVELOPING.md
Comment (haq204) [inline DEVELOPING.md:258]: maybe go into detail on how to run tests and lint. getting the full environment setup to run tests locally is somewhat involved.

Esp because as of right now community members can't run CI

**Review by haq204 — COMMENTED** [review]
just some minor editing comments on DEVELOPING.md but otherwise looks fine with me

File: DEVELOPING.md
Comment (LanceEa) [reply DEVELOPING.md:258]: Since this is more less a quick check list I will point at the current testing section and add a note about running `make lint`.

If we want to improve the instructions in those sections, we can PR those separate.

**Comment by LanceEa** [comment]
@haq204 - I addressed your feedback and rebased on master. So, once this is green it should be good to go.

**Review by haq204 — APPROVED** [review]

**Pull request merged**
